src/main/java/com/gibgab/service/controller/FeedController.java

```diff
@@ -0,0 +1,37 @@
+package com.gibgab.service.controller;
+
+import com.gibgab.service.database.entity.ApplicationUser;
+import com.gibgab.service.database.entity.Post;
+import com.gibgab.service.database.repository.PostRepository;
+import com.gibgab.service.database.repository.UserRepository;
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.http.HttpStatus;
+
+import java.util.Optional;
+import java.util.List;
+
+@RestController
+public class FeedController {
+
+    @Autowired
+    private Environment env;
+
+    @Autowired
+    private PostRepository postRepository;
+
+    final int POSTS_PER_PAGE = 10;
+
+    @GetMapping({"/feed"})
+    public @ResponseBody List<Post> get_feed(@PathVariable Optional<Integer> page) {
+        return postRepository.findByOrderByIdDesc(PageRequest.of(0, POSTS_PER_PAGE));
```

Comment:
Is the Optional `page` not intended to be used here? Might want to remove it from the signature, or make the page request use it.